storybook/src/stories/UseDeepCompareEffect/API/API.tsx

```diff
@@ -0,0 +1,40 @@
+import throttle from 'lodash.throttle'
+import React, { useRef, useState } from 'react'
+import { useDeepCompareEffect } from 'too-many-hooks'
+import COUNTRY_CODES from './countryCodes.json'
+
+interface Props {
+  firstName: string
+  countryKey: keyof typeof COUNTRY_CODES
+}
+
+const API: React.FC<Props> = ({ firstName, countryKey }) => {
+  const [predictedAge, setPredictedAge] = useState<string>()
+
+  const getThrottledPrediction = useRef(
```

Comment:
What's the advantage of making this a `useRef` instead of a `useCallback`?

Reply:
It needs to be in a ref so the value does not change between renders since it is wrapped in the throttle function, which makes `useCallback` mad that it cannot infer the dependencies. To do `useCallback` it then needs to redo the props type and do an arrow function to the throttle and then you cannot just call the throttled function directly. Plus it is just less clean than using the ref since it will never change anyway. 

<img width="559" alt="image" src="https://user-images.githubusercontent.com/39521214/215371088-df432f25-8599-44ec-b355-a773d19b0511.png">